File: spec/factories/email_notification.rb
# typed: false
# frozen_string_literal: true

FactoryBot.define do
  factory :email_notification do
    user
    unsubscription_key { "#{SecureRandom.uuid}-#{SecureRandom.uuid}" }
    event_followed_user { true }
    event_liked_episode_record { true }
    event_next_season_came { true }
    event_favorite_works_added { true }
    event_related_works_added { true }
  end
end

File: go/internal/model/email_notification.go
package model

import "time"

// EmailNotification はユーザーのメール通知設定のドメインエンティティ
type EmailNotification struct {
	ID                      EmailNotificationID
	UserID                  UserID
	UnsubscriptionKey       string
	EventFollowedUser       bool
	EventLikedEpisodeRecord bool
	CreatedAt               time.Time
	UpdatedAt               time.Time
}
